src/ralph/backends/data/es.py

```diff
@@ -0,0 +1,397 @@
+            time alive.
+    """
+
+    id: Optional[str]
+    keep_alive: Optional[str]
+
+
+class ESQuery(BaseQuery):
+    """Elasticsearch query model.
+
+    Attributes:
+        query (dict): A search query definition using the Elasticsearch Query DSL.
+            See Elasticsearch search reference for query DSL syntax:
+            https://www.elastic.co/guide/en/elasticsearch/reference/8.9/search-search.html#request-body-search-query
+        query_string (str): The Elastisearch query in the Lucene query string syntax.
+            See Elasticsearch search reference for Lucene query syntax:
+            https://www.elastic.co/guide/en/elasticsearch/reference/8.9/search-search.html#search-api-query-params-q
+        pit (dict): Limit the search to a point in time (PIT). See ESQueryPit.
+        size (int): The maximum number of documents to yield.
+        sort (str or list): Specify how to sort search results. Set to `_doc` or
+            `_shard_doc` if order doesn't matter.
+            See https://www.elastic.co/guide/en/elasticsearch/reference/8.9/sort-search-results.html
+        search_after (list): Limit search query results to values after a document
+            matching the set of sort values in `search_after`. Used for pagination.
+        track_total_hits (bool): Number of hits matching the query to count accurately.
+            Not used. Always set to `False`.
+    """  # pylint: disable=line-too-long # noqa: E501
+
+    query: dict = {"match_all": {}}
+    pit: ESQueryPit = ESQueryPit()
+    size: Optional[int]
+    sort: Union[str, List[dict]] = "_shard_doc"
+    search_after: Optional[list]
+    track_total_hits: Literal[False] = False
+
+
+class ESDataBackend(BaseDataBackend):
+    """Elasticsearch data backend."""
+
+    name = "es"
+    query_model = ESQuery
+    settings_class = ESDataBackendSettings
+
+    def __init__(self, settings: settings_class = None):
+        """Instantiate the Elasticsearch client.
+
+        Args:
+            settings (ESDataBackendSettings or None): The data backend settings.
+                If `settings` is `None`, a default settings instance is used instead.
+        """
+        self.settings = settings if settings else self.settings_class()
+        self._client = None
+
+    @property
+    def client(self):
+        """Create an Elasticsearch client if it doesn't exist."""
+        if not self._client:
+            self._client = Elasticsearch(
+                self.settings.HOSTS, **self.settings.CLIENT_OPTIONS.dict()
+            )
+        return self._client
+
+    def status(self) -> DataBackendStatus:
+        """Check Elasticsearch cluster connection and status."""
+        try:
+            self.client.info()
+        except ESConnectionError as error:
+            logger.error("Failed to connect to Elasticsearch: %s", error)
+            return DataBackendStatus.AWAY
+
+        cluster_status = self.client.cat.health()
```

Comment:
Do we need to handle the possibility that Elasticsearch becomes unavailable just after the `self.client.info()` request?

Reply:
If it becomes unavailable, we reach the end of the method where we log and return a `DataBackendStatus.ERROR`, no?